File: 0344-reverse-string/solution.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        
        int i=0,n=s.size()-1,temp;
        
        while(i<n){
            temp = s[i];
            s[i] = s[n];
            s[n] = temp;
            n--;
            i++;
        }
        
    }
};

File: 0344-reverse-string/solution.java
class Solution {
    public void reverseString(char[] s) {
        int ptr = 0, ptr2 = s.length-1;
        while(ptr < ptr2){
            s[ptr] = (char)(s[ptr] ^ s[ptr2]);
            s[ptr2] = (char)(s[ptr] ^ s[ptr2]);
            s[ptr] = (char)(s[ptr] ^ s[ptr2]);
            ptr++;
            ptr2--;
        }      
    }
}
